webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/GoogleCloudDataVault.scala

```diff
@@ -72,6 +73,18 @@ class GoogleCloudDataVault(uri: URI, credential: Option[GoogleServiceAccountCred
     } yield (bytes, encoding)
   }
 
+  override def listDirectory(path: VaultPath)(implicit ec: ExecutionContext): Fox[List[VaultPath]] = {
+    val objName = path.toUri.getPath.tail
+    val blobs = storage.list(bucket,
+                             Storage.BlobListOption.prefix(objName),
+                             Storage.BlobListOption.currentDirectory(),
+                             Storage.BlobListOption.pageSize(MAX_EXPLORED_ITEMS_PER_LEVEL))
+    val subDirectories = blobs.getValues.asScala.toList.filter(_.isDirectory)
```

Comment:
The limit should probably be applied after this isDirectory filtering. Maybe use a default pageSize (1000?). This should not be super expensive. The exploring itself will be.

Reply:
I used the default page size limit here (removed the `Storage.BlobListOption.pageSize(MAX_EXPLORED_ITEMS_PER_LEVEL)`)  option